spec.html

```diff
@@ -38101,11 +38103,12 @@ <h1>NotifyWaiter ( _WL_, _W_ )</h1>
         <h1>AtomicReadModifyWrite ( _typedArray_, _index_, _value_, _op_ )</h1>
         <p>The abstract operation AtomicReadModifyWrite takes arguments _typedArray_, _index_, _value_, and _op_ (a read-modify-write modification function). _op_ takes two List of byte values arguments and returns a List of byte values. This operation atomically loads a value, combines it with another value, and stores the result of the combination. It returns the loaded value. It performs the following steps when called:</p>
         <emu-alg>
-          1. Let _buffer_ be ? ValidateSharedIntegerTypedArray(_typedArray_).
+          1. Let _buffer_ be ? ValidateIntegerTypedArray(_typedArray_).
           1. Let _i_ be ? ValidateAtomicAccess(_typedArray_, _index_).
           1. Let _arrayTypeName_ be _typedArray_.[[TypedArrayName]].
           1. If _typedArray_.[[ContentType]] is ~BigInt~, let _v_ be ? ToBigInt(_value_).
           1. Otherwise, let _v_ be ? ToInteger(_value_).
+          1. If IsDetachedBuffer(_buffer_) is *true*, throw a *TypeError* exception.
```

Comment:
Isn't this redundant? `ValidateIntegerTypedArray` already calls `ValidateTypedArray`, and that performs the detached buffer check.

Ditto most of the other new `IsDetachedBuffer` checks.

Reply:
Ah, no, it's because ? ToInteger() calls ToNumber, which could call ToPrimitive, which could call arbitrary code. :(

Reply:
Uuuugh.

That's worth adding `NOTE`s for, I think, especially since an implementation which thinks the same thing I did could plausibly end up with a security bug. E.g. after each of these surprisingly-necessary steps, add a

> 1. NOTE: The above check is not redundant with the check in ValidateIntegerTypedArray because the call to `ToInteger` on the preceding line can have arbitrary side effects, which could cause the buffer to become detached after the call to ValidateIntegerTypedArray.